storage/src/metadata/mod.rs

```diff
@@ -515,4 +521,75 @@ mod tests {
             metadata.destroy().await.unwrap();
         });
     }
+
+    #[test_traced]
+    fn test_diffs() {
+        // Initialize the deterministic context
+        let executor = deterministic::Runner::default();
+        executor.start(|context| async move {
+            // Create a metadata store
+            let cfg = Config {
+                partition: "test".to_string(),
+            };
+            let mut metadata: Metadata<_, U64> =
+                Metadata::init(context.clone(), cfg).await.unwrap();
+
+            // Put initial keys
+            for i in 0..100 {
+                metadata.put(U64::new(i), vec![i as u8; 100]);
+            }
+
+            // First sync - should write everything to the first blob
+            //
+            // 100 keys * (8 bytes for key + 4 bytes for len + 100 bytes for value) + 8 bytes for version + 4 bytes for checksum
+            metadata.sync().await.unwrap();
+            let buffer = context.encode();
+            assert!(buffer.contains("skipped_total 0"), "{}", buffer);
+            assert!(
+                buffer.contains("runtime_storage_write_bytes_total 11212"),
+                "{}",
+                buffer
```

Comment:
```suggestion
                "{buffer}"
```

There are a bunch of places in this PR where we move vars into `{}`. Should we do that here, too?

Reply:
Same for other uses of `"{}"`

Reply:
I just fixed the cases where the linter complained. Will fix here too.